src/types.ts

```diff
@@ -55,8 +55,13 @@ export class ClientError extends Error {
   }
 }
 
+export type MaybeFunction<T> = T | (() => T);
+
 export type RequestDocument = string | DocumentNode
 
+export type PatchedRequestInit = Omit<Dom.RequestInit, "headers">
```

Comment:
Couldnt think of a more elegant way.

ie. it's not possible to 
```
interface PatchedRequestInit extends Dom.RequestInit {
  headers?: () => Dom.RequestInit['headers'] | Dom.RequestInit['headers']
}
```
because that "doesn't extend the interface correctly".
I also tried `Dom.RequestInit & { headers?: () => Dom.RequestInit['headers'] }` but that didn't work either